File: app/src/main/java/com/joshuahalvorson/safeyoutube/view/network/YoutubeDataClient.java
package com.joshuahalvorson.safeyoutube.view.network;

import com.joshuahalvorson.safeyoutube.view.model.PlaylistResultOverview;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface YoutubeDataClient {
    public static final String BASE_URL = "https://www.googleapis.com/youtube/v3/";

    @GET("playlistItems")
    Call<PlaylistResultOverview> getPlaylistOverview(@Query("part") String part,
                                                     @Query("playlistId") String playlistId,
                                                     @Query("maxResults") String maxResults,
                                                     @Query("key") String key);
}

File: app/src/main/java/com/joshuahalvorson/locklist/network/YoutubeDataClient.kt
package com.joshuahalvorson.locklist.network

import com.joshuahalvorson.locklist.model.Models.PlaylistResultOverview
import com.joshuahalvorson.locklist.model.Models.VideoInfo
import io.reactivex.Flowable
import retrofit2.http.GET
import retrofit2.http.Query

interface YoutubeDataClient {
    @GET("playlistItems")
    fun getPlaylistOverview(@Query("part") part: String,
                            @Query("playlistId") playlistId: String,
                            @Query("maxResults") maxResults: String,
                            @Query("key") key: String): Flowable<PlaylistResultOverview>

    @GET("playlists")
    fun getPlaylistInfo(@Query("part") part: String,
                        @Query("id") playlistId: String,
                        @Query("key") key: String): Flowable<PlaylistResultOverview>

    @GET("videos")
    fun getVideoInfo(@Query("part") part: String,
                     @Query("id") playlistId: String,
                     @Query("key") key: String): Flowable<VideoInfo>

}